sky/clouds/kubernetes.py

```diff
@@ -0,0 +1,353 @@
+
+        r = resources
+        acc_dict = self.get_accelerators_from_instance_type(r.instance_type)
+        if acc_dict is not None:
+            custom_resources = json.dumps(acc_dict, separators=(',', ':'))
+        else:
+            custom_resources = None
+
+        # resources.memory and cpus are None if they are not explicitly set.
+        # We fetch the default values for the instance type in that case.
+        cpus, mem = self.get_vcpus_mem_from_instance_type(
+            resources.instance_type)
+        return {
+            'instance_type': resources.instance_type,
+            'custom_resources': custom_resources,
+            'region': region.name,
+            'cpus': str(cpus),
+            'memory': str(mem),
+            'timeout': str(self.TIMEOUT),
+            'k8s_ssh_key_secret_name': self.SKY_SSH_KEY_SECRET_NAME,
+        }
+
+    def get_feasible_launchable_resources(self,
+                                          resources: 'resources_lib.Resources'):
+        if resources.use_spot:
+            return ([], [])
+        fuzzy_candidate_list: List[str] = []
+        if resources.instance_type is not None:
+            assert resources.is_launchable(), resources
+            resources = resources.copy(accelerators=None)
+            return ([resources], fuzzy_candidate_list)
+
+        def _make(instance_list):
+            resource_list = []
+            for instance_type in instance_list:
+                r = resources.copy(
+                    cloud=Kubernetes(),
+                    instance_type=instance_type,
+                    accelerators=None,
+                )
+                resource_list.append(r)
+            return resource_list
+
+        # Currently, handle a filter on accelerators only.
+        accelerators = resources.accelerators
+        if accelerators is None:
+            # Return a default instance type with the given number of vCPUs.
+            default_instance_type = Kubernetes.get_default_instance_type(
+                cpus=resources.cpus,
+                memory=resources.memory,
+                disk_tier=resources.disk_tier)
+            if default_instance_type is None:
+                return ([], [])
+            else:
+                return (_make([default_instance_type]), [])
+
+        assert len(accelerators) == 1, resources
+        # If GPUs are requested, return an empty list.
+        # TODO(romilb): Add GPU support.
+        return ([], [])
+
+    @classmethod
+    def check_credentials(cls) -> Tuple[bool, Optional[str]]:
+        if os.path.exists(os.path.expanduser(_CREDENTIAL_PATH)):
+            # Test using python API
+            return kubernetes_utils.check_credentials()
+        else:
+            return False, 'Credentials not found - ' \
+                          f'check if {_CREDENTIAL_PATH} exists.'
+
+    def get_credential_file_mounts(self) -> Dict[str, str]:
+        return {_CREDENTIAL_PATH: _CREDENTIAL_PATH}
```

Comment:
Ah yeah, that's right - but I think `check_credentials()` should guard against marking k8s as enabled on the spot controller because it includes a reachability check (it tries to run a simple API call on the k8s API server, and if it fails it marks it as not disabled). Do you think that would be enough?